src/updater/updater.cpp

```diff
@@ -120,10 +120,16 @@ auto mmapFile(const std::string &filename, boost::interprocess::mode_t mode)
     using boost::interprocess::file_mapping;
     using boost::interprocess::mapped_region;
 
+    {
+        storage::io::FileReader file(filename, storage::io::FileReader::VerifyFingerprint);
```

Comment:
Opening this file with the `FileReader` just for verifying the fingerprint seems a bit clumsy. On Windows this fails in a way since a subsequent call from `file_mapping` results in `Access denied`.

Is there a way to read just the Fingerprint from the `mapped_region`? Should we implement an explicit `Close` function for the `FileReader`/`FileWriter`, or do an implicit `fclose` on the destructor?